docs/source/configuration/configuration_basics.md

```diff
@@ -159,6 +182,27 @@ conf_loader = OmegaConfigLoader(conf_source=conf_path)
 conf_catalog = conf_loader["catalog"]
 ```
 
+### How to load a data catalog with credentials in code?
+Assuming your project contains a catalog and credentials file each located in a `base` and `local` environment respectively, you can use the `OmegaConfigLoader` to load these configurations and then pass them on to a `DataCatalog` object to get access to the catalog entries with resolved credentials.
```

Comment:
In this section can we please put a `Note` admonition that explains why this shouldn't be done within a node, nodes should be pure functions with no knowledge of IO. Then mention hooks should be the only way to manipulate things like the catalog at runtime.

Reply:
I think that's a very good point and imo related to whether this is now in the right place in the docs. How do we distinguish between using "Kedro as a library" just in notebooks for exploration vs. as part of their Kedro projects where they should be following best practices? Any thoughts on this @stichbury and @astrojuanlu ?

Reply:
Is the issue that people cannot use hooks in notebook exploration so have to put this in a node when they shouldn't? Because I'd think that is probably "OK" in as much as the code is not production-quality and it's not ideal but won't be world-ending. To prevent people doing this in nodes in production code projects when they shouldn't, a simple warning as Joel suggests should be sufficient, right? With a nod to the fact that "if you're unable to use hooks because you're doing exploratory notebook code" then we  see you, and don't forget to refactor your code when you move it into a project for best practice.

Reply:
I'm working back from the many times we see users who do this inline within a node- my suggestion would be anytime we mention this pattern we should also say it's only to be used in situation X and not Y

Reply:
I don't think there is any issue in repeating this in multiple places either

Reply:
> Then mention hooks should be the only way to manipulate things like the catalog at runtime.

Hooks have a hard time manipulating the catalog at runtime https://github.com/kedro-org/kedro/issues/2728 better to leave that out of the note

> How do we distinguish between using "Kedro as a library" just in notebooks for exploration vs. as part of their Kedro projects where they should be following best practices?

I think it doesn't harm to include a reminder that node functions should indeed not do any I/O? This is not only a Kedro thing, but goes well beyond https://rhodesmill.org/brandon/slides/2015-05-pywaw/hoist/

Reply:
I've added a note, lmk if it's sufficient.

Reply:
Love it! thanks @merelcht. I don't have approve / resolve rights anymore so take this as that!